File: chroma-league/src/main/java/com/bonepl/chromaleague/hud/parts/resource/MordekaiserShieldBar.java
package com.bonepl.chromaleague.hud.parts.resource;

import com.bonepl.chromaleague.hud.parts.ProgressBar;
import com.bonepl.chromaleague.state.GameStateHelper;
import com.bonepl.razersdk.animation.AnimatedFrame;
import com.bonepl.razersdk.animation.Color;
import com.bonepl.razersdk.animation.Frame;

public class MordekaiserShieldBar extends AnimatedFrame {
    @Override
    public Frame getFrame() {
        addAnimationFrame(new ProgressBar(ResourceBars.getResourceBarKeys(), GameStateHelper.getResourcePercentage(), Color.WHITE));
        return super.getFrame();
    }
}

File: src/hud/parts/resource/MordekaiserShieldBar.ts
import {AnimatedFrame} from '../../../razer-sdk/animation/AnimatedFrame.js';
import {Frame} from '../../../razer-sdk/animation/Frame.js';
import {StaticColor} from '../../../razer-sdk/color/StaticColor.js';
import {ProgressBar} from '../ProgressBar.js';
import {getResourceBarKeys} from './ResourceBars.js';
import {getResourcePercentage} from '../../../state/GameStateHelper.js';

export class MordekaiserShieldBar extends AnimatedFrame {
    pollFrame(): Frame {
        this.addAnimationFrame(new ProgressBar(getResourceBarKeys(), getResourcePercentage(), StaticColor.WHITE));
        return super.pollFrame();
    }
}
